### ISSUE TITLE
ptipython raises ArgumentError: argument 1: <type 'exceptions.TypeError'>: wrong type

### ISSUE THREAD

**Issue by zhoushun123** [jonathanslenders/python-prompt-toolkit#131]
**My python environment:**
windows 7 sp1 64bit
Python 2.7.10 (default, May 23 2015, 09:44:00) [MSC v.1500 64 bit (AMD64)]
IPython 3.1.0 
prompt-toolkit-0.40 
ptpython-0.14

**Traceback:**
```python
---------------------------------------------------------------------------
ArgumentError                             Traceback (most recent call last)
D:\Python27\lib\runpy.pyc in _run_module_as_main(mod_name, alter_argv)
    160         sys.argv[0] = fname
    161     return _run_code(code, main_globals, None,
--> 162                      "__main__", fname, loader, pkg_name)
    163
    164 def run_module(mod_name, init_globals=None,

D:\Python27\lib\runpy.pyc in _run_code(code, run_globals, init_globals, mod_name
, mod_fname, mod_loader, pkg_name)
     70                        __loader__ = mod_loader,
     71                        __package__ = pkg_name)
---> 72     exec code in run_globals
     73     return run_globals
     74

D:\Python27\Scripts\ptipython.exe\__main__.py in <module>()
      7 if __name__ == '__main__':
      8     sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
----> 9     sys.exit(run())

D:\Python27\lib\site-packages\ptpython\entry_points\run_ptipython.pyc in run()
     62               history_filename=os.path.join(config_dir, 'history'),
     63               configure=configure,
---> 64               user_ns=user_ns)
     65
     66

D:\Python27\lib\site-packages\ptpython\ipython.pyc in embed(**kwargs)
    235         config.InteractiveShellEmbed = config.TerminalInteractiveShell
    236         kwargs['config'] = config
--> 237     shell = InteractiveShellEmbed.instance(**kwargs)
    238     initialize_extensions(shell, config['InteractiveShellApp']['extensio
ns'])
    239     shell(header=header, stack_depth=2, compile_flags=compile_flags)

D:\Python27\lib\site-packages\IPython\config\configurable.pyc in instance(cls, *
args, **kwargs)
    335         # Create and save the instance
    336         if cls._instance is None:
--> 337             inst = cls(*args, **kwargs)
    338             # Now make sure that the instance will also be returned by
    339             # parent classes' _instance attribute.

D:\Python27\lib\site-packages\ptpython\ipython.pyc in __init__(self, *a, **kw)
    176             return self.user_ns
    177
--> 178         self._eventloop = create_eventloop()
    179         ipython_input = IPythonInput(
    180             self,

D:\Python27\lib\site-packages\prompt_toolkit\shortcuts.pyc in create_eventloop()

     69         from prompt_toolkit.eventloop.posix import PosixEventLoop as Loo
p
     70
---> 71     return Loop()
     72
     73

D:\Python27\lib\site-packages\prompt_toolkit\eventloop\win32.pyc in __init__(sel
f)
     26     def __init__(self):
     27         self._event = _create_event()
---> 28         self._console_input_reader = ConsoleInputReader()
     29         self._calls_from_executor = []
     30

D:\Python27\lib\site-packages\prompt_toolkit\terminal\win32_input.pyc in __init_
_(self)
     95
     96     def __init__(self):
---> 97         self.handle = windll.kernel32.GetStdHandle(STD_INPUT_HANDLE)
     98
     99     def read(self):

ArgumentError: argument 1: <type 'exceptions.TypeError'>: wrong type
```

**Comment by jonathanslenders** [comment]
Hi @zhoushun123, I'm sorry for the late reply.
Does this still happen? (With: prompt-toolkit-0.42  ptpython-0.16)

Maybe STD_INPUT_HANDLE should be a c_ulong rather than a HANDLE. Could you try replacing HANDLE with ctypes.c_ulong at the following three lines?
https://github.com/jonathanslenders/python-prompt-toolkit/blob/master/prompt_toolkit/win32_types.py#L5

**Comment by zhoushun123** [comment]
Yes,this still happen with prompt-toolkit-0.42 ptpython-0.16 (with the latest ipython 3.2.1).I made the change in your way,and it works.Will you fix this in the future release?

**Comment by jonathanslenders** [comment]
Fixed in https://github.com/jonathanslenders/python-prompt-toolkit/commit/8ac0c8936ff815ed7bb48c5e8d81c59cfeddc24b

I will try to do a new release today or tomorrow.

**Comment by zhoushun123** [comment]
oh yeah,it's been fixed in the lastest prompt-toolkit-0.43 and ptpython-0.18
